packages/uploadthing/src/internal/handler.ts

```diff
@@ -233,119 +233,169 @@ export const buildRequestHandler = <
     }
 
     try {
-      const { files, input: userInput } = reqBody as {
-        files: string[];
-        input: Json;
-      };
+      if (actionType === "upload") {
+        const { files, input: userInput } = (await req.json()) as {
+          files: string[];
+          input: Json;
+        };
 
-      // validate the input
-      let parsedInput: Json = {};
-      try {
-        const inputParser = uploadable._def.inputParser;
-        parsedInput = await getParseFn(inputParser)(userInput);
-      } catch (error) {
-        return new UploadThingError({
-          code: "BAD_REQUEST",
-          message: "Invalid input",
-          cause: error,
-        });
-      }
+        let parsedInput: Json = {};
 
-      let metadata: Json = {};
-      try {
-        metadata = await uploadable._def.middleware({
+        // validate the input
+        try {
+          const inputParser = uploadable._def.inputParser;
+          parsedInput = await getParseFn(inputParser)(userInput);
+        } catch (error) {
+          console.error(error);
+          return new UploadThingError({
+            code: "BAD_REQUEST",
+            message: "Invalid input",
+            cause: error,
+          });
+        }
+
+        const metadata = await uploadable._def.middleware({
```

Comment:
This can throw errors

Reply:
so it should be wrapped in a try-catch?

Reply:
Like it was before, and then return the UploadThing error

Reply:
The whole switch is wrapped in a try catch that does this? 
![image](https://github.com/pingdotgg/uploadthing/assets/11494384/80138bdb-d317-442c-afb8-978a32e817d8)

Reply:
Ah, mobile diff view is hard to parse 😅

That doesnt quite make sense though does it? There are more things that can fail in that big block which would cause a sub-optimal error message?

Reply:
Yeah... I should really go through and clean up this logic....